test/test_kubeclient.rb

```diff
@@ -232,6 +232,37 @@ def test_get_all
     assert_instance_of(Kubeclient::Namespace, result['namespace'][0])
   end
 
+  def test_api_bearer_token_success
+    stub_request(:get, 'http://localhost:8080/api/v1beta3/pods')
+      .with(headers: { Authorization: 'Bearer valid_token' })
+      .to_return(body: open_test_json_file('pod_list_b3.json'),
+                 status: 200)
+
+    client = Kubeclient::Client.new 'http://localhost:8080/api/'
+    client.bearer_token('valid_token')
+
+    pods = client.get_pods
+
+    assert_equal('Pod', pods.kind)
+    assert_equal(1, pods.size)
+  end
+
+  def test_api_bearer_token_failure
+    error_message = '"/api/v1beta3/pods" is forbidden because ' \
+                    'system:anonymous cannot list on pods in'
+
+    stub_request(:get, 'http://localhost:8080/api/v1beta3/pods')
+      .with(headers: { Authorization: 'Bearer invalid_token' })
```

Comment:
> Shouldn't that be covered by existing test cases? It will just be a replica of test_api_bearer_token_failure but with no bearer token header or bearer_token call...

well, you tell me :)
It depends on  the implementation, some systems might treat invalid token and lack of tocken in the same way, some treat them differently. if k8s treats it the same way, then the above is good enough.